qa/tasks/cephfs/test_admin.py

```diff
@@ -742,6 +743,309 @@ def test_fs_rename_fails_with_mirroring_enabled(self):
+                                self.fs2.id: self.fs2.name}
+
+        self.mount_a.remount(cephfs_name=self.fs1.name)
+        self.mount_b.remount(cephfs_name=self.fs2.name)
+
+        self.captester = CapTester()
+        self.captester.write_test_files((self.mount_a, self.mount_b))
+
+        self.mount_a.umount_wait()
+        self.mount_b.umount_wait()
+
+    def tearDown(self):
+        self.mount_a.umount_wait()
+        self.mount_b.umount_wait()
+        self.fs.mon_manager.run_cluster_cmd(
+            args=f'auth rm {self.client_name}')
+        super(TestFsSwap, self).tearDown()
+
+    def remount_fss(self):
+        moncap = gen_mon_cap_str((("rw", self.fs1.name),
+                                  ("rw", self.fs2.name)))
+        osdcap = gen_osd_cap_str((("rw", self.fs1.name),
+                                  ("rw", self.fs2.name)))
+        mdscap = gen_mds_cap_str((("rw", self.fs1.name),
+                                  ("rw", self.fs2.name)))
+        self.fs1.mon_manager.run_cluster_cmd(args='auth add '
+            f'{self.client_name} mon "{moncap}" osd "{osdcap}" '
+            f'mds "{mdscap}"')
+
+        keyring = self.fs.mon_manager.get_keyring(self.client_id) + '\n'
+        keyring_path_a = self.mount_a.client_remote.mktemp(data=keyring)
+        keyring_path_b = self.mount_b.client_remote.mktemp(data=keyring)
+
+        self.mount_a.mount(client_id=self.client_id,
+                           client_keyring_path=keyring_path_a,
+                           cephfs_mntpt='/', cephfs_name=self.fs1.name)
+        self.mount_b.mount(client_id=self.client_id,
+                           client_keyring_path=keyring_path_b,
+                           cephfs_mntpt='/', cephfs_name=self.fs2.name)
+
+    def run_rw_tests(self):
+        self.captester.conduct_pos_test_for_read_caps()
+        self.captester.conduct_pos_test_for_write_caps()
+
+    def _check_fs_name_on_fs_pools(self, fss):
+        for fs in fss:
+            for pool in fs.get_data_pool_names(True):
+                self.check_pool_application_metadata_key_value(pool, 'cephfs',
+                    'data', fs.name)
+            self.check_pool_application_metadata_key_value(
+                fs.get_metadata_pool_name(), 'cephfs', 'metadata', fs.name)
+
+    def _are_fsnames_and_fscids_together(self):
+        '''
+        Are FS names and FSIDs together on same the FS as they were before
+        running "ceph fs swap" command?
+        '''
+        fs1_id_swapped = self.orig_fs_id_name[self.fs1.id] == self.fs1.name
+        fs2_id_swapped = self.orig_fs_id_name[self.fs2.id] == self.fs2.name
+
+        if fs1_id_swapped and fs2_id_swapped:
+            return True
+        elif not fs1_id_swapped and not fs2_id_swapped:
+            return False
+        else:
+            raise RuntimeError(
+                'Unexpected situation occured: FSID for one FS swapped but '
+                'not for the other FS.')
+
+    def test_fs_swap(self):
```

Comment:
Include a short description of the test.

Reply:
Isn't it redundant? It tests exactly what the method name says.

Reply:
No, it isn't.

```
def test_fs_swap():
    """
    that the file system name and other attributes (pool tags, etc...) are exchanged - but not FSCID.
    """
```

Reply:
Ok.

Reply:
Done.